src/base_alloc/base_alloc.c

```diff
@@ -278,7 +278,11 @@ void umf_ba_free(umf_ba_pool_t *pool, void *ptr) {
     umf_ba_chunk_t *chunk = (umf_ba_chunk_t *)ptr;
 
     utils_mutex_lock(&pool->metadata.free_lock);
-    assert(pool_contains_pointer(pool, ptr));
+    if (!pool_contains_pointer(pool, ptr)) {
```

Comment:
why should we remove this `assert`. It was able to catch the issue #903. With this PR we just disable it and issue become silent.